apps/laboratory/tests/smart-account.spec.ts

```diff
@@ -13,32 +13,59 @@ testModalSmartAccount.beforeEach(async ({ modalValidator }) => {
   await modalValidator.expectConnected()
 })
 
-testModalSmartAccount('it should sign with eoa', async ({ modalPage, modalValidator }) => {
-  await modalPage.sign()
-  await modalPage.approveSign()
-  await modalValidator.expectAcceptedSign()
+testModalSmartAccount('it should use a Smart Account', async ({ modalPage, modalValidator }) => {
+  const walletModalPage = modalPage as ModalWalletPage
+  const walletModalValidator = modalValidator as ModalWalletValidator
+
+  await modalValidator.expectConnected()
+  await walletModalPage.openAccount()
+  await walletModalValidator.expectActivateSmartAccountPromoVisible(false)
+
+  await walletModalPage.openProfileView()
+  await walletModalPage.openSettings()
+  await walletModalValidator.expectChangePreferredAccountToShow(EOA)
+  await walletModalPage.closeModal()
+})
+
+testModalSmartAccount('it should sign with 6492', async ({ modalPage, modalValidator }) => {
+  const walletModalPage = modalPage as ModalWalletPage
+  const walletModalValidator = modalValidator as ModalWalletValidator
+
+  await walletModalPage.sign()
+  await walletModalPage.approveSign()
+  await walletModalValidator.expectAcceptedSign()
+
+  const signature = await walletModalPage.getSignature()
+  const address = await walletModalPage.getAddress()
+  const chainId = await walletModalPage.getChainId()
+
+  await walletModalValidator.expectValidSignature(signature, address, chainId)
 })
 
 testModalSmartAccount(
-  'it should switch to its smart account and sign',
+  'it should switch to its eoa and sign',
```

Comment:
And this one says switch to it's eoa instead of SA, but in the test we are expecting it to be switched to SA

Reply:
nono, the expect is for what the button should show, so if we expect the button to show "Switch to your EOA" means you are on your SA